pglogical_functions.c

```diff
@@ -600,6 +601,23 @@ pglogical_drop_subscription(PG_FUNCTION_ARGS)
 				LWLockRelease(PGLogicalCtx->lock);
 				break;
 			}
+
+			if (apply->proc->waitLock)
+			{
+				const LOCKTAG apply_lock_tag = apply->proc->waitLock->tag;
+				if (((LockTagType) apply_lock_tag.locktag_type == LOCKTAG_TRANSACTION) &&
```

Comment:
This check works only for 2 processes deadlock cycle (apply worker and drop sub process).
PostgreSQL deadlock detection ( `DeadLockCheck(PGPROC  *proc)` ) is not working here because drop subscription worker do not locked by apply worker, it just waiting for apply being stopped.